src/resolvers.ts

```diff
@@ -8,28 +8,40 @@ import { promisify } from 'bluebird';
 import { ResolverMap, Config } from './types';
 import { accessNestedObject } from './utils';
 
+const stackCache: Map<string, any> = new Map();
+
+// Single shared instantiation of aws-sdk clients - exported for test access
+export const cft = new AWS.CloudFormation();
+export const asm = new AWS.SecretsManager();
+
 export const resolvers: ResolverMap = {
   cft: async (argument: string, config: Config) => {
-    if (!AWS.config.region) {
-      AWS.config.update({ region: config.awsRegion });
-    }
-    const cft = new AWS.CloudFormation();
     const parsedArgument = argument.split('.', 2);
-    let stack: DescribeStacksOutput;
-    try {
-      stack = await cft.describeStacks({ StackName: parsedArgument[0] }).promise();
-    } catch (e) {
-      console.warn(
-        `Could not get info for stack with name ${parsedArgument[0]} when parsing cft reference ${argument}: ${e}`
-      );
-      return undefined;
-    }
-    if (stack.Stacks.length == 0) {
-      console.warn(`Could not locate stack with name ${parsedArgument[0]} when parsing cft reference ${argument}`);
-      return undefined;
+
+    let stack: Stack = stackCache.get(parsedArgument[0]);
+
+    if (!stack) {
+      if (!AWS.config.region) {
+        AWS.config.update({ region: config.awsRegion });
+      }
+      let describeStack: DescribeStacksOutput;
+      try {
+        describeStack = await cft.describeStacks({ StackName: parsedArgument[0] }).promise();
+      } catch (e) {
+        console.warn(
+          `Could not get info for stack with name ${parsedArgument[0]} when parsing cft reference ${argument}: ${e}`
+        );
+        return undefined;
+      }
+      if (describeStack.Stacks.length == 0) {
+        console.warn(`Could not locate stack with name ${parsedArgument[0]} when parsing cft reference ${argument}`);
+        return undefined;
+      }
+      stack = describeStack.Stacks[0];
+      stackCache.set(parsedArgument[0], stack);
     }
 
-    for (const output of stack.Stacks[0].Outputs) {
+    for (const output of stack.Outputs) {
```

Comment:
This is complex enough that you can pull it out into its own file that wraps over the AWS commands.  `resolvers` being a single file was bound to eventually not scale.

Reply:
This will also allow you to refactor this method to be a little terser, if desired.

Something like:
```
   let stack: Stack = stackCache.get(parsedArgument[0]);
   if (!stack) {
      stack = retrieveStackFromAWS(...);
   }
   ....
}